File: src/editor/gui/display/CardJList.java
package editor.gui.display;

import java.util.ArrayList;
import java.util.List;

import javax.swing.DefaultListModel;
import javax.swing.DefaultListSelectionModel;
import javax.swing.JList;

import editor.database.card.Card;
import editor.gui.SettingsDialog;

/**
 * This class represents an element that can display a list of cards by name.
 * It cannot be edited and does not support selection.
 * 
 * @author Alec Roelke
 */
@SuppressWarnings("serial")
public class CardJList extends JList<String>
{
	/**
	 * List of cards to display.
	 */
	private List<Card> cards;
	
	/**
	 * Create an empty CardJList.
	 */
	public CardJList()
	{
		this(new ArrayList<>());
	}
	
	/**
	 * Create a new CardJList displaying the given list of cards.
	 * 
	 * @param c card list to display
	 */
	public CardJList(List<Card> c)
	{
		super();
		cards = c;
		setSelectionModel(new DefaultListSelectionModel()
		{
			{
				setSelectionMode(SINGLE_SELECTION);
			}
			
			@Override
			public void setSelectionInterval(int index0, int index1)
			{
				super.setSelectionInterval(-1, -1);
			}
		});
		setModel(new DefaultListModel<String>()
		{
			@Override
			public String getElementAt(int index)
			{
				return cards.get(index).unifiedName();
			}
			
			@Override
			public int getSize()
			{
				return cards.size();
			}
		});
		setVisibleRowCount(SettingsDialog.getAsInt(SettingsDialog.EXPLICITS_ROWS));
	}
	
	/**
	 * Set the list of cards to display.
	 * 
	 * @param c New list to display
	 */
	public void setCards(List<Card> c)
	{
		cards = c;
	}
}

File: src/main/scala/editor/gui/display/CardJList.scala
package editor.gui.display

import editor.database.card.Card
import editor.gui.settings.SettingsDialog

import javax.swing.DefaultListModel
import javax.swing.DefaultListSelectionModel
import javax.swing.JList
import javax.swing.ListSelectionModel

/**
 * UI element for displaying a list of cards.  Does not support selection.
 * 
 * @constructor create a new card list showing a list of cards
 * @param cards cards to initially display; can be changed later
 * 
 * @author Alec Roelke
 */
class CardJList(var cards: Seq[Card] = Seq.empty) extends JList[String] {
  setSelectionModel(new DefaultListSelectionModel {
    setSelectionMode(ListSelectionModel.SINGLE_SELECTION)
    override def setSelectionInterval(index0: Int, index1: Int) = super.setSelectionInterval(-1, -1)
  })
  setModel(new DefaultListModel[String] {
    override def getElementAt(index: Int) = cards(index).unifiedName
    override def getSize = cards.size
  })
  setVisibleRowCount(SettingsDialog.settings.editor.categories.explicits)
}
